src/main/java/org/kohsuke/github/GitHubClient.java

```diff
@@ -409,12 +401,28 @@ private void logRequest(@Nonnull final GitHubRequest request) {
     }
 
     @Nonnull
-    protected abstract GitHubResponse.ResponseInfo getResponseInfo(GitHubRequest request) throws IOException;
-
-    protected abstract void handleLimitingErrors(@Nonnull GitHubResponse.ResponseInfo responseInfo) throws IOException;
+    protected ResponseInfo getResponseInfo(GitHubRequest request) throws IOException {
+        return connector.getResponseInfo(request, authorizationProvider);
+    }
```

Comment:
I chose to treat the request as unmodifiable, whereas your `GitHubClient` has `private GitHubRequest prepareRequest(GitHubRequest request) throws IOException {`. In that method you add the auth headers, and also do some accept/encoding stuff that I don't understand very well. Probably my PR is just plain missing those features.

I think this way of handling authorization by passing the provider vs modifying the request is the only substantive difference between the two PRs, and I like your way better.